### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by yosraemad**
## What?

Created Login Page and Sign-up page

## Why?

To be able to register new users and sign in.

## How?

- created an auth service that handles all api requests
- used redux toolkit to save user state and interacts with the auth service.

## Notes

I still didn't unencrypt the token, I will do this later when we implement features that need the role and so on

### CODE REVIEW

File: src/pages/Login/components/SignInForm/SignInForm.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const [loading, setLoading] = useState(false);
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setLoading(true);
+    dispatch(login({ email, password }))
+      .unwrap()
+      .then(() => {
```
Comment (omarralaa) [inline src/pages/Login/components/SignInForm/SignInForm.js:25]: Remove logic from UI Component

File: src/pages/SignUp/components/SignUpForm/SignUpForm.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import styles from "./SignUpForm.module.css";
+import Login from "../Login/Login";
+import { Navigate, useNavigate } from "react-router-dom";
+import { register } from "../../../../slices/auth";
+import { useDispatch } from "react-redux";
+
+const SignUpForm = () => {
+  const navigate = useNavigate();
+
+  const [firstName, setFirstName] = useState("");
```
Comment (omarralaa) [inline src/pages/SignUp/components/SignUpForm/SignUpForm.js:14]: if there is time, replace with suitable form library

File: src/services/auth_service.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import axios from "axios";
+
+const API_URL = "http://localhost:9000/api/v1/auth/";
```
Comment (omarralaa) [inline src/services/auth_service.js]: move to config file

File: src/slices/message.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {};
+
+const messageSlice = createSlice({
```
Comment (omarralaa) [inline src/slices/message.js:5]: Fill or remove

**Review by omarralaa — CHANGES_REQUESTED** [review]

**Comment by yosraemad** [comment]
.env.local file should be placed inside the root directory of the app, you can find it on our drive

**Pull request merged**